File: productComputerC/BladeStr.cpp
/*
 * BladeStr.cpp
 *
 *  Created on: 21.11.2011
 *      Author: christian
 */

#include "BladeStr.h"

BladeStr::BladeStr() {
}

BladeStr::~BladeStr() {
}

BladeStr::BladeStr(const string& baseVector)
{
	prefactor = 1;
	this->baseVectors.push_back(baseVector);
}

BladeStr::BladeStr(const vector<string>& baseVectors)
{
	prefactor = 1;
	this->baseVectors = baseVectors;
}

BladeStr::BladeStr(const float prefactor, const string& baseVector)
{
	this->prefactor = prefactor;
	this->baseVectors.push_back(baseVector);
}

BladeStr::BladeStr(const float prefactor, const vector<string>& baseVectors)
{
	this->prefactor = prefactor;
	this->baseVectors = baseVectors;
}

void BladeStr::print() {
	std::cout << prefactor << "[";
	for (vector<string>::const_iterator it = baseVectors.begin(); it != baseVectors.end(); ++it)
		std::cout << *it << ",";

	std::cout << "]";
}

File: tba/src/main/java/de/gaalop/productComputer/BladeStr.java
package de.gaalop.productComputer;

import java.util.Arrays;

/**
 * Blade represented by an array of strings
 * @author Christian Steinmetz
 */
public class BladeStr {

    private float prefactor;
    private String[] baseVectors;

    public BladeStr(String baseVector) {
        this.prefactor = 1;
        this.baseVectors = new String[]{baseVector};
    }

    public BladeStr(String[] baseVectors) {
        this.prefactor = 1;
        this.baseVectors = baseVectors;
    }

    public BladeStr(float prefactor, String baseVector) {
        this.prefactor = prefactor;
        this.baseVectors = new String[]{baseVector};
    }

    public BladeStr(float prefactor, String[] baseVectors) {
        this.prefactor = prefactor;
        this.baseVectors = baseVectors;
    }

    public String[] getBaseVectors() {
        return baseVectors;
    }

    public float getPrefactor() {
        return prefactor;
    }

    public void setBaseVectors(String[] baseVectors) {
        this.baseVectors = baseVectors;
    }

    public void setPrefactor(float prefactor) {
        this.prefactor = prefactor;
    }

    @Override
    public String toString() {
        return prefactor+Arrays.toString(baseVectors);
    }

}
